etcdserver/api/v3rpc/key.go

```diff
@@ -29,25 +32,127 @@ func New(s etcdserver.V3DemoServer) pb.EtcdServer {
 }
 
 func (h *handler) Range(ctx context.Context, r *pb.RangeRequest) (*pb.RangeResponse, error) {
+	if err := checkRangeRequest(r); err != nil {
+		return &pb.RangeResponse{}, err
```

Comment:
It's basically just like normal Go idioms. If it's nil, then the client will get a nil response. If you're returning an error, then the client should check the error and not use the response.